src/modal.ts

```diff
@@ -0,0 +1,201 @@
+import { setSameWidth, setElementDisplay } from "./utils.js";
+import { resetDropdownItem, initDropdownSelect, generateTimeDropdown, populateTimeDropdowns } from "./inputs.js";
+import { getModal, getModalInputById, getModalInputs } from "./selectors.js";
+import { removeUnsavedEventTile, styleEventTile, updateTileTime } from "./eventTile.js";
+import { saveEvent, getTime, getEventLength, getEndTime } from "./eventsData.js";
+import { currentEventTileId, emptyEventTitle, modalInputsIds, modalTitleId } from "./constants.js";
+import { currentEventDataKey, storeDataInLocalStorage, getDataFromLocalStorage } from "./handleLocalStorage.js";
+import { getDisplayableTime, hourIsValid, minutesAreValid } from "./timeCalculations.js";
+import { EventData } from "./types/main.js";
+
+const setTimeDateInputWidths = (modal: HTMLElement): void => {
+    Object.entries(modalInputsIds).forEach(([key, inputId]) => {
+        const btn: HTMLElement = modal.querySelector(`#${inputId}-btn`);
+        if (btn) {
+            btn.style.width = "fit-content";
+            setSameWidth(`#${inputId}-btn`, `#${inputId}`);
+        }
+    });
+};
+
+const removeSeparators = (modal: HTMLElement): void => {
+    modal
+        .querySelectorAll(".with-separator")
+        .forEach((item: HTMLElement) => item.classList.remove("separated-top", "separated-bottom"));
+};
+
+const resetModal = (modal: HTMLElement): void => {
+    modal.querySelectorAll(".preview-settings").forEach((item: HTMLElement) => setElementDisplay(item, "flex"));
+    modal.querySelectorAll(".full-settings").forEach((item: HTMLElement) => item.classList.remove("active"));
+    modal.querySelectorAll(".select-input").forEach((dropdownItem: HTMLElement) => resetDropdownItem(dropdownItem));
+    removeUnsavedEventTile();
+    setTimeDateInputWidths(modal);
+    removeSeparators(modal);
+};
+
+const addSeparators = (parent: HTMLElement): void => {
+    const parentClasses = parent.classList;
+    const [prevDiv, nextDiv] = [parent.previousElementSibling as HTMLElement, parent.nextElementSibling as HTMLElement];
```

Comment:
You are doing thing here called "sviestas sviestuotas" and also trying to "hack" the type system with casting.

Here you are experiencing common development problem that some element can be of time `T` (in your case `Element`) or null.

So in this case you should do one of following:
1. Stop the execution if `previousElementSibling` or `nextElementSibling` is null
2. Throw an error that `previousElementSibling` and `nextElementSibling` are required in order to add separators